### ISSUE TITLE
SocialShare Plugin

### ISSUE THREAD

**Issue by jenstornell** [jenstornell/kirby-plugin-wishlist#14]
SocialShare Plugin - a plugin to make it trivial to add 'Share' social media buttons to a page - e.g., 'Share on Facebook', 'Share on Twitter', 'Share on LinkedIn', etc. @luxlogica

**Comment by jenstornell** [comment]
A starting point: http://sharingbuttons.io/

**Comment by flokosiol** [comment]
I think I'll give it a try to wrap this into a plugin.

**Comment by jenstornell** [comment]
@flokosiol I was about to wrap it as a plugin a while ago myself, but when I thought more about it, I did not see the benefits of it anymore. I will tell you why,..

Let's say it's wrapped as a plugin. The css and the js is added to the plugin. Then the instructions for the plugin may looks like this in the template header:

```php
<?= css('[plugin_path]/socialmedia.css'); ?>
<?= js('[plugin_path]/socialmedia.js'); ?>
```

Then it's almost no difference compared to just add them to the assets folder. It's just different paths.

*Personally I use another way for css and js ([Bricks](https://github.com/jenstornell/kirby-bricks)) where I merge and minify the files with Gulp to only use one http request on each of them. A plugin for it would in my case actually make it more difficult.*

I'm not going to stop you from developing this. I just wanted to share my thoughts from where I was about to wrap it as a plugin myself. Maybe you have another idea around it? 😄

**Comment by luxlogica** [comment]
Ok, here is the use-case, which may be the most flexible, and also create the least amount of work...:

At the moment, I give my clients the option of adding "share on social media" buttons to their sites, usually by adding checkbox fields to their Panel, like this:

<img width="761" alt="screen shot 2017-05-12 at 4 09 39 pm" src="https://cloud.githubusercontent.com/assets/2994472/25985313/783a5bfe-372d-11e7-946d-13668d4f91bb.png">

When the client ticks the boxes, we'll add the 'share' button to the appropriate place in the site. And that is the "boring" bit for me, as a developer. Every time I want to add one of these, I go to [SharingButtons](http://sharingbuttons.io), and copy code from there. Although this is a great site, it's a hassle doing the copying every time: they try to do too much (even creating a button icon) and every time we need to tick settings, analyse the code and copy just the bits we need into our template.

What would be *really* useful for us would be if we could just use a plugin function, or even kirbytag, to generate these automatically for us - but enabling us to use our own icons/buttons, etc. - along the lines of...:

```
(socialshare type:facebook icon:facebook title:share on Facebook)
```
...where "icon" and "title" would be optional parameters. This would output something like:

```
<a 
    title="share on Facebook"
    target="_blank" 
    href="https://facebook.com/sharer/sharer.php?u=<?php echo esc($page->url(),'url')?>">
    <span class="fa fa-facebook"></span>
</a>
```

**Comment by flokosiol** [comment]
@jenstornell Thanks for sharing your thoughts. I already thought about an easier way to (automatically) add the assets. But there is not other/better way … and I think it's good the way it is, because the developer can control how everything is implemented.

@luxlogica Great to hear what's your usecase. 

Wrapping this into a plugin may also be helpful for less experienced developers. I will think about it …

**Comment by jenstornell** [comment]
@luxlogica Yes,I agree that checkboxes in the panel can be useful for clients. To build it, extending a checkboxes field would probably be the most simple way.

https://getkirby.com/docs/cheatsheet/panel-fields/checkboxes
https://github.com/jenstornell/kirby-boiler-field/blob/master/fields/boilercheckboxes/boilercheckboxes.php

About the kirbytext tag. It means that every page would have a custom set of social buttons. What is the case for that? Often they are set globally and in the template we only decide if social buttons should be included or not.

**Comment by luxlogica** [comment]
@jenstornell: we need to have control over the appearance/design of the buttons, as well as where they appear. 

In some sites the buttons should just be the plain logo, in others they may show as circular buttons, or as square buttons with a label. In some sites, the 'social share' buttons might be placed in the footer, while in others, it may be in the site menu (using a different style), or within the 'Contact' section. In a Blog Section, you might want to include it at the bottom of every Blog Article (with custom styling to fit the design of the page). And in some sites, you may use *all* of these options at once! ;-)

**Comment by flokosiol** [comment]
Here we go … first version … poorly tested :wink:
https://github.com/flokosiol/kirby-sharingbuttons

Would love to get some feedback from you.
